sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java

```diff
@@ -248,53 +248,40 @@ public void setSessionFactory(SessionFactory sessionFactory) {
   }
 
   @Override
-  public Collection<SampleIdentity> getIdentitiesByExternalNameOrAliasPartialMatch(String externalNames) throws IOException {
-    if (isStringEmptyOrNull(externalNames)) return Collections.emptySet();
+  public Collection<SampleIdentity> getIdentitiesByExternalNameOrAliasAndProject(String query, Long projectId, boolean exactMatch)
+      throws IOException {
+    if (isStringEmptyOrNull(query)) return Collections.emptySet();
     @SuppressWarnings("unchecked")
-    List<SampleIdentity> records = (List<SampleIdentity>) SampleIdentityImpl.getSetFromString(externalNames)
-        .stream().map(extName -> {
-          String str = DbUtils.convertStringToSearchQuery(extName, false);
+    Set<SampleIdentity> records = (Set<SampleIdentity>) SampleIdentityImpl.getSetFromString(query)
+        .stream().map(extNameOrAlias -> {
+          String str = DbUtils.convertStringToSearchQuery(extNameOrAlias, false);
           Criteria criteria = currentSession().createCriteria(SampleIdentityImpl.class);
+          if (projectId != null) {
+            criteria.add(Restrictions.eq("project.id", projectId));
+          }
           criteria.add(Restrictions.or(Restrictions.ilike("externalName", str), Restrictions.ilike("alias", str)));
           return criteria.list();
-        }).flatMap(list -> list.stream()).collect(Collectors.toList());
-    return records;
-  }
-
-  @Override
-  public Collection<SampleIdentity> getIdentitiesByExactExternalName(String externalNames) throws IOException {
-    if (externalNames == null) return Collections.emptyList();
-
-    Collection<SampleIdentity> partialMatches = getIdentitiesByExternalNameOrAliasPartialMatch(externalNames);
-    return filterOnlyExactExternalNameMatches(partialMatches, externalNames);
-  }
-
-  @Override
-  public Collection<SampleIdentity> getIdentitiesByExactExternalNameAndProject(String externalNames, Long projectId) throws IOException {
-    if (isStringEmptyOrNull(externalNames)) return Collections.emptySet();
-    if (projectId == null) throw new IllegalArgumentException("Must provide a projectId in search");
-    @SuppressWarnings("unchecked")
-    Set<SampleIdentity> records = (Set<SampleIdentity>) SampleIdentityImpl.getSetFromString(externalNames)
-        .stream().map(extName -> {
-          String str = DbUtils.convertStringToSearchQuery(extName, false);
-          Criteria criteria = currentSession().createCriteria(SampleIdentityImpl.class);
-          criteria.add(Restrictions.eq("project.id", projectId));
-          criteria.add(Restrictions.ilike("externalName", str));
-          return criteria.list();
-        }).flatMap(list -> list.stream()).distinct().collect(Collectors.toSet());
+        }).flatMap(list -> list.stream())
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)